File: src/chapters/2_sorting/array/bubble_sort/bubble.pas
program BubbleSort;

uses crt;

const
  N = 12;
  AInit: array[1..N] of Integer = (5, 99, 101, 100, 2, 90, 9, 1, 2, 22, 5, 6);

var
  A: array[1..N] of Integer;
  i, j, Temp: Integer;

begin
  clrscr;

  for i := 1 to N do
    A[i] := AInit[i];

  for i := 1 to N - 1 do
    for j := 1 to N - i do
      if A[j] > A[j + 1] then
      begin
        (* xor swap *)
        A[j]      := A[j] Xor A[j +1];
        A[j + 1]  := A[j] Xor A[j +1];
        A[j]      := A[j] Xor A[j +1];
      end;

  WriteLn('Sorted array in ascending order:');
  for i := 1 to N do
    Write(A[i], ' ');
    
  ReadLn;
end.

File: src/chapters/2_sorting/array/bubble_sort/bubble.go
package main

import "fmt"

func bubbleSort(arr []int) {
	n := len(arr)
	for i := 0; i < n-1; i++ {
		for j := 0; j < n-i-1; j++ {
			if arr[j] > arr[j+1] {
				arr[j], arr[j+1] = arr[j+1], arr[j]
			}
		}
	}
}

func main() {
	numbers := []int{5, 99, 101, 100, 2, 90, 9, 1, 2, 22, 5, 6}
	fmt.Println("Unsorted array:", numbers)
	bubbleSort(numbers)
	fmt.Println("Sorted array:", numbers)
}

